### DISCUSSION TITLE
Discover, stream and download LibriVox audiobooks.

### DISCUSSION THREAD

**Discussion by Mesum-Hussain** [PaulWoitaschek/Voice#1658] [Ideas]
I am imagining three tabs, **Home**, **Local** and **LibriVox**. In the **Local** tab, the app will function as it is functioning currently: play the books from the "Audiobooks" folder. But in the **LibriVox** tab, the user should be able to discover, stream and download the LibriVox audiobooks. The download button should add the downloaded LibriVox audiobook to the "Audiobooks" folder.

We can stream them from the [Internet Archive](https://archive.org/details/librivoxaudio) or maybe directly from the [LibriVox website](https://librivox.org/). Maybe RSS would be useful.

A **Home** tab is also needed to display the 'currently in progress' books regardless of where they are from (LibriVox or the Local Storage), so that the user can continue listening very easily.

A Floating Action Button (FAB) for **Search** would really be helpful for searching books across both the Local Storage and the LibriVox library.

For the **LibriVox** tab, we can take inspiration from various audiobook streaming apps.....we can suggest similar books and the books from the same author.

**Comment by Vuizur** [comment]
I think this is a great idea. I don't know about the best way of implementing the GUI, here the most important thing is probably that it is not too difficult, but the download of the books itself should not be a huge challenge.

For example, this Python code I quickly tested downloads the first Spanish book download as "adventure":

```python
import internetarchive

# Also search in description "Read in Spanish"
search = internetarchive.search_items('subject:"adventure" collection:librivoxaudio description:"Read in Spanish"')
for result in search:
    print(result['identifier'])
    #Metadata
    item = internetarchive.get_item(result['identifier'])
    print(item.metadata)
    # Download the zip file with the 64kbps mp3 files
    result = item.download(formats='64KBPS MP3', glob_pattern='*.mp3', destdir='output_dir')

    break
```
(Ok, it only downloads the first chapter because of some bug I couldn't figure out, but still.)
In Kotlin this would have to be achieved using the REST API, but that should be possible I think.

So this way we could get the most important features: search/language filter/maybe genre filter.

There also exists an Librivox API, but it is unmaintained and doesn't seem to support language filtering, at least I couldn't figure it out that fast.
